x/distribution/spec/03_begin_block.md

```diff
@@ -0,0 +1,80 @@
+
+The amount of proposer reward is calculated from pre-commits Tendermint messages in order to incentivize validators to wait and include additional pre-commits in the block. All provision rewards are added to a provision reward pool which validator holds individually (`ValidatorDistribution.ProvisionsRewardPool`).
+
+## The Distribution Scheme
+
+See [params](07_params.md) for description of parameters.
+
+Let `fees` be the total fees collected in the previous block. All fees are
+collected in a specific module account during the block. During `BeginBlock`,
+they are sent to the `"distribution"` `ModuleAccount`. No other sending of
+tokens occur. Instead, the rewards each account is entitled to are stored, and
+withdrawals can be triggered through the messages `FundCommunityPool`,
+`WithdrawValidatorCommission` and `WithdrawDelegatorReward`.
+
+### Reward to the Community Pool
+
+The community pool gets `communitytax * fees`, plus any remaining dust after
+validators get their rewards, which will always be rounded down to nearest
+integer value.
+
+### Reward To the Validators
+
+The proposer will receive a base reward of `fees * baseproposerreward`. In
+addition, they will receive a bonus of `fees * bonusproposerreward * P`, where
+`P = (precommits included / total bonded validator power)`. The more precommits
+the propser includes, the larger `P` will be. It can never be larger than `1.00`
+, and will always be larger than `2/3`.
+
+Any remaining fees are distributed among all the bonded validators, including
+the proposer, in proportion to their consensus power.
+
+```
+powFrac = validator power / total bonded validator power
+proposerMul = baseproposerreward + bonusproposerreward * P
+voteMul = 1 - communitytax - proposerMul
+```
+
+In total, the proposer will receive `fees * voteMul * powFrac + fees * proposerMul`.
+All other validators will receive `fees * voteMul`.
+
+### Rewards to Delegators
+
+Each validators rewards are distributed to its delegators. Note that the
+validator also has a self-delegation.
+
+The validator sets a commission rate. The commission rate is flexible, but each
+validator sets a max rate which may never be exceeded, and a max daily increase
+which may never be exceeded. This protects delegators against a validator
+suddenly increasing their commission rate, taking all the rewards.
```

Comment:
```suggestion
Each validator's rewards are distributed to its delegators. The
the validator also has a self-delegation.

The validator sets a commission rate. The commission rate is flexible, but each
validator sets a maximum rate and a maximum daily increase. These maximums cannot be exceeded and protect delegators from sudden increases of validator commission rates to prevent validators from taking all of the rewards.
```